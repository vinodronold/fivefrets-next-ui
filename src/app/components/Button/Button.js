import React from 'react'
import { bool, node } from 'prop-types'
import { color } from '../../constants/theme'

const _propTypes = {
  disabled: bool,
  children: node.isRequired
}
const _defaultProps = {
  disabled: false
}

const Button = ({ disabled, children, ...props }) => (
  <button disabled={disabled} {...props}>
    {children}
    <style jsx>{`
      button {
        font-size: 0.938rem;
        cursor: pointer;
        padding: 0.5rem 1rem;
        letter-spacing: 0.2rem;
        line-height: 0.938rem;
        height: 2.5rem;
        background: transparent;
        border: none;
        transition: all 0.5s;
      }
      button:focus {
        outline: none;
      }
      button:disabled {
        cursor: not-allowed;
      }
    `}</style>
    <style jsx>{`
      button {
        color: ${!disabled && color.primary()};
      }
      button:hover {
        color: ${!disabled && color.secondary()};
      }
    `}</style>
  </button>
)
Button.propTypes = _propTypes
Button.defaultProps = _defaultProps

export default Button
