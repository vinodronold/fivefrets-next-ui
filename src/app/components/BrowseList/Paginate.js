import React from 'react'
import { number, func } from 'prop-types'
import Button from '../Button'

const _propTypes = {
  curr_page: number,
  total_pages: number,
  MoveToBrowsePage: func.isRequired
}

const _defaultProps = {
  curr_page: 1,
  total_pages: 1
}

/// prop type validation is not happening in test

const Paginate = ({ curr_page, total_pages, MoveToBrowsePage }) => (
  <div>
    <Button
      disabled={curr_page === 1}
      onClick={() => {
        MoveToBrowsePage(curr_page - 1)
      }}>
      &#8606; Prev
    </Button>

    <Button
      onClick={() => {
        MoveToBrowsePage(1)
      }}>{`${curr_page} of ${total_pages}`}</Button>

    <Button
      disabled={curr_page === total_pages}
      onClick={() => {
        MoveToBrowsePage(curr_page + 1)
      }}>
      Next &#8608;
    </Button>

    <style jsx>{`
      div {
        padding: 1rem;
        display: flex;
        justify-content: center;
        flexwrap: wrap;
      }
    `}</style>
  </div>
)

Paginate.propTypes = _propTypes
Paginate.defaultProps = _defaultProps

export default Paginate
