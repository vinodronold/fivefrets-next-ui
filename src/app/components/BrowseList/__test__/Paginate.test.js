import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Paginate from '../Paginate'
import Button from '../../Button'

describe('components/BrowseList/Paginate', () => {
  it('should match the snapshot', () => {
    const _paginate = renderer.create(<Paginate curr_page={1} total_pages={10} MoveToBrowsePage={jest.fn()} />)
    expect(_paginate).toMatchSnapshot()
  })

  it('should go to Page 1 when Prev is clicked', () => {
    const _paginate = shallow(<Paginate curr_page={2} total_pages={10} MoveToBrowsePage={jest.fn()} />)
    _paginate.find(Button).at(0).simulate('click')
    expect(_paginate).toMatchSnapshot()
  })

  it('should go to Page 2 when Next is clicked', () => {
    const _paginate = shallow(<Paginate curr_page={2} total_pages={10} MoveToBrowsePage={jest.fn()} />)
    _paginate.find(Button).at(2).simulate('click')
    expect(_paginate).toMatchSnapshot()
  })

  it('should go to Page 1 when Page Number is clicked', () => {
    const _paginate = shallow(<Paginate curr_page={2} total_pages={10} MoveToBrowsePage={jest.fn()} />)
    _paginate.find(Button).at(1).simulate('click')
    expect(_paginate).toMatchSnapshot()
  })
})
