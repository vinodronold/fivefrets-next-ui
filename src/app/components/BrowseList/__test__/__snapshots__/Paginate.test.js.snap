// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BrowseList/Paginate should go to Page 1 when Page Number is clicked 1`] = `
<div
  className="jsx-2366590580"
>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    ↞ Prev
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    2 of 10
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    Next ↠
  </Button>
  <JSXStyle
    css="div.jsx-2366590580{padding:1rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flexwrap:wrap;-ms-flexwrap:wrap;flexwrap:wrap;}"
    styleId="2366590580"
  />
</div>
`;

exports[`components/BrowseList/Paginate should go to Page 1 when Prev is clicked 1`] = `
<div
  className="jsx-2366590580"
>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    ↞ Prev
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    2 of 10
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    Next ↠
  </Button>
  <JSXStyle
    css="div.jsx-2366590580{padding:1rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flexwrap:wrap;-ms-flexwrap:wrap;flexwrap:wrap;}"
    styleId="2366590580"
  />
</div>
`;

exports[`components/BrowseList/Paginate should go to Page 2 when Next is clicked 1`] = `
<div
  className="jsx-2366590580"
>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    ↞ Prev
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    2 of 10
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    Next ↠
  </Button>
  <JSXStyle
    css="div.jsx-2366590580{padding:1rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flexwrap:wrap;-ms-flexwrap:wrap;flexwrap:wrap;}"
    styleId="2366590580"
  />
</div>
`;

exports[`components/BrowseList/Paginate should match the snapshot 1`] = `
<div
  className="jsx-2366590580"
>
  <button
    className="jsx-856601377 jsx-1171890091 "
    disabled={true}
    onClick={[Function]}
  >
    ↞ Prev
  </button>
  <button
    className="jsx-856601377 jsx-3822723886 "
    disabled={false}
    onClick={[Function]}
  >
    1 of 10
  </button>
  <button
    className="jsx-856601377 jsx-3822723886 "
    disabled={false}
    onClick={[Function]}
  >
    Next ↠
  </button>
</div>
`;
